beater/server.go

```diff
@@ -62,12 +62,18 @@ func newServer(config Config, publish successCallback) *http.Server {
 	}
 }
 
-func start(server *http.Server, ssl *SSLConfig) {
-	if ssl.isEnabled() {
-		go server.ListenAndServeTLS(ssl.Cert, ssl.PrivateKey)
-	} else {
-		go server.ListenAndServe()
-	}
+func start(server *http.Server, ssl *SSLConfig) chan(error) {
+	c := make(chan(error), 1)
```

Comment:
The advantage of the above is that we don't need to pass around an error channel and the it is directly visible in the `beater.go` on what is happening. 

For testing you could also extract that part in a method so you can test it separately. 

In general the part I don't like about the previous code is that we start go routines in two different places for kind of the same thing.